xds/src/main/java/io/grpc/xds/LookasideLb.java

```diff
@@ -248,6 +196,215 @@ private static ManagedChannel initLbChannel(
     return channel;
   }
 
+  /**
+   * A LoadBalancerProvider that provides a load balancer with a fixed XdsConfig.
+   */
+  private final class FixedXdsConfigBalancerProvider extends LoadBalancerProvider {
+    final XdsConfig xdsConfig;
+    final XdsConfig oldXdsConfig;
+    final EndpointWatcher oldEndpointWatcher;
+
+    FixedXdsConfigBalancerProvider(XdsConfig xdsConfig) {
+      this.xdsConfig = xdsConfig;
+      oldXdsConfig = LookasideLb.this.xdsConfig;
```

Comment:
I found this very wired/confusing/awkward. Now that your design is "config-fixed load balancer". Why would there be something as "old config"? The class encapsulation is broken, it violates the idea of defining such a class. It should be the outer class that memorizes the previous config and this inner class should just reference that.

Reply:
>Now that your design is "config-fixed load balancer". Why would there be something as "old config"?

The old config is the config of the old config-fixed load balancer.

>The class encapsulation is broken, it violates the idea of defining such a class.

This is a field of outer class. Inner class accessing outer class is not encapsulation broken. On the contrary, modifying a private field of inner class by outer class is encapsulation broken.

Reply:
I see that `oldXdsConfig` is a field of inner class and NOT of outer class. But I am okay with this implementation.

Reply:
> `LookasideLb.this.xdsConfig` is a field of outer class. Inner class accessing outer class field is not encapsulation broken. On the contrary, modifying a private field of inner class by outer class is encapsulation broken.

That's not my point. My point is, the config-fixed load balancer is a balancer that bounded to one config, it shouldn't aware its sibling's config, which is what the inner class's `oldXdsConfig` field mean. IMO, the outer class should manage the switching between two `FixedXdsConfigBalancer` instances instead of letting one `FixedXdsConfigBalancer` look at what its sibling's config was.

Reply:
I agree it's better to let `FixedXdsConfigBalancer` not to remember old config. However, 

- you still need to `watchEndpointData()` with new watcher in the inner class `FixedXdsConfigBalancer`, so it's better to call `cancelEndpointDataWatch()` right next to it instead of in the outer class, which would be more clear to understand.
- `oldXdsConfig` is also used for inner class to decide whether to create a new channel.